jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -404,6 +405,13 @@ public void addIncludedPaths(String... pathspecs)
     @Override
     protected void doStart() throws Exception
     {
+        if (_deflaterPool == null)
+        {
+            _deflaterPool = getServer().getBean(DeflaterPool.class);
+            if (_deflaterPool == null)
+                _deflaterPool = new DeflaterPool(DEFAULT_POOL_CAPACITY, getCompressionLevel(), true);
```

Comment:
The pool capacity needs to be configurable.
I think default should be 64, which should cover the most number of cores ever likely to be seen for a while.

You then need a setter on the handler and all the usual stuff in xml and mod files to allow it to be set when the module is enabled.

@sbordet @joakime Hmmm why do we actually need a capacity? couldn't we just let it grow to the maximum required?

@lachlan-roberts can you modify the class so that a capacity of 0 is no pool, but a capacity of -1 is an unlimited pool and we can make that the default.